src/server/analytic_services/prometheus_reporting.js

```diff
@@ -69,6 +69,12 @@ async function start_server(
         return;
     }
     const metrics_request_handler = async (req, res) => {
+        // TODO: This is a temporary condition to avoid the token authentication for metrics, 
+        // Need to add authentication for NSFS NC also after confirming flow with the Scale team
+        if (!process.env.NC_NSFS_NO_DB_ENV && config.METRICS_AUTH_ENABLED) {
+            // Authorize bearer token metrics endpoint
+            http_utils.authorize_bearer(req, res, [ "metrics-auth", "admin" ]);
```

Comment:
I would rename the role to just "metrics" because that "-auth" doesn't make sense as a role name.